src/Altinn.App.Core/Helpers/UserHelper.cs

```diff
@@ -50,55 +49,50 @@ public UserHelper(
-            }
-            else if (claim.Type.Equals(AltinnCoreClaimTypes.UserId, StringComparison.Ordinal))
+            User = context.User,
+            UserName = tokenClaims.GetValueOrDefault(AltinnCoreClaimTypes.UserName),
+            UserId = tokenClaims.GetValueOrDefault(AltinnCoreClaimTypes.UserId) switch
             {
-                userContext.UserId = Convert.ToInt32(claim.Value, CultureInfo.InvariantCulture);
-            }
-            else if (claim.Type.Equals(AltinnCoreClaimTypes.PartyID, StringComparison.Ordinal))
+                { } value => Convert.ToInt32(value, CultureInfo.InvariantCulture),
+                _ => throw new Exception("Could not get user profile - could not retrieve user ID from claims"),
+            },
+            PartyId = tokenClaims.GetValueOrDefault(AltinnCoreClaimTypes.PartyID) switch
             {
-                userContext.PartyId = Convert.ToInt32(claim.Value, CultureInfo.InvariantCulture);
-            }
-            else if (claim.Type.Equals(AltinnCoreClaimTypes.AuthenticationLevel, StringComparison.Ordinal))
+                { } value => Convert.ToInt32(value, CultureInfo.InvariantCulture),
+                _ => default,
+            },
+            AuthenticationLevel = tokenClaims.GetValueOrDefault(AltinnCoreClaimTypes.AuthenticationLevel) switch
             {
-                userContext.AuthenticationLevel = Convert.ToInt32(claim.Value, CultureInfo.InvariantCulture);
-            }
-        }
-
-        if (userContext.UserId == default)
-        {
-            throw new Exception("Could not get user profile - could not retrieve user ID from claims");
-        }
+                { } value => Convert.ToInt32(value, CultureInfo.InvariantCulture),
+                _ => default,
+            },
+        };
 
         UserProfile userProfile =
             await _profileClient.GetUserProfile(userContext.UserId)
             ?? throw new Exception("Could not get user profile while getting user context");
-        userContext.UserParty = userProfile.Party;
 
-        if (context.Request.Cookies[_settings.GetAltinnPartyCookieName] != null)
-        {
-            userContext.PartyId = Convert.ToInt32(
-                context.Request.Cookies[_settings.GetAltinnPartyCookieName],
-                CultureInfo.InvariantCulture
-            );
-        }
+        if (partyCookieValue is not null)
+            userContext.PartyId = Convert.ToInt32(partyCookieValue, CultureInfo.InvariantCulture);
 
         if (userContext.PartyId == userProfile.PartyId)
-        {
             userContext.Party = userProfile.Party;
-        }
-        else
-        {
+        else if (userContext.PartyId > 0)
             userContext.Party = await _altinnPartyClientService.GetParty(userContext.PartyId);
-        }
+
+        userContext.UserParty = userProfile.Party;
+
+        userContext.SocialSecurityNumber =
+            userContext.Party?.SSN ?? userContext.Party?.Person?.SSN ?? userContext.UserParty?.SSN;
```

Comment:
it would suck if `userContext.Party?.SSN == ""` 😄

Reply:
Agreed. Very sucky... I don't know the likelyhood of all these edge cases, but perhaps I should check for empty and whitespace?